eras/alonzo/impl/cddl-files/alonzo.cddl

```diff
@@ -111,297 +341,188 @@ script_data_hash = $hash32 ; New
-;   - the value of costmdls map at key 0 is encoded as an indefinite length
+;   - the value of cost_models map at key 0 is encoded as an indefinite length
 ;     list and the result is encoded as a bytestring. (our apologies)
 ;   - the language ID tag is also encoded twice. first as a uint then as
 ;     a bytestring. (our apologies)
-;
-; Note that each Plutus language represented inside a transaction must have
-; a cost model in the costmdls protocol parameter in order to execute,
-; regardless of what the script integrity data is. In the Alonzo era,
-; this means costmdls must have a key 0 for Plutus V1.
-;
-; Finally, note that in the case that a transaction includes datums but does not
-; include any redeemers, the script data format becomes (in hex):
-; [ 80 | datums | A0 ]
-; corresponding to a CBOR empty list and an empty map (our apologies).
-
-; address = bytes
-; reward_account = bytes
+; 
+; Note that each Plutus language represented inside a transaction
+; must have a cost model in the cost_models protocol parameter in
+; order to execute, regardless of what the script integrity data
+; is. In the Alonzo era, this means cost_models must have a key 0
+; for Plutus V1.
+; 
+; Finally, note that in the case that a transaction includes
+; datums but does not include any redeemers, the script data
+; format becomes (in hex):
+;   [ 80 | datums | A0 ]
+; 
+; corresponding to a CBOR empty list and an empty map (our
+; apologies).,
+; 
+; NEW:
+;   script_data_hash
+; 
+script_data_hash = $hash32
 
-; address format:
-; [ 8 bit header | payload ];
-;
-; shelley payment addresses:
-; bit 7: 0
-; bit 6: base/other
-; bit 5: pointer/enterprise [for base: stake cred is keyhash/scripthash]
-; bit 4: payment cred is keyhash/scripthash
-; bits 3-0: network id
-;
-; reward addresses:
-; bits 7-5: 111
-; bit 4: credential is keyhash/scripthash
-; bits 3-0: network id
-;
-; byron addresses:
-; bits 7-4: 1000
-
-; 0000: base address: keyhash28,keyhash28
-; 0001: base address: scripthash28,keyhash28
-; 0010: base address: keyhash28,scripthash28
-; 0011: base address: scripthash28,scripthash28
-; 0100: pointer address: keyhash28, 3 variable length uint
-; 0101: pointer address: scripthash28, 3 variable length uint
-; 0110: enterprise address: keyhash28
-; 0111: enterprise address: scripthash28
-; 1000: byron address
-; 1110: reward account: keyhash28
-; 1111: reward account: scripthash28
-; 1001 - 1101: future formats
-
-certificate =
-  [ stake_registration
-  // stake_deregistration
-  // stake_delegation
-  // pool_registration
-  // pool_retirement
-  // genesis_key_delegation
-  // move_instantaneous_rewards_cert
-  ]
+; To compute a script hash, note that you must prepend
+; a tag to the bytes of the script before hashing.
+; The tag is determined by the language.
+; The tags in the Conway era are:
+;   "\\x00\" for multisig scripts
+;   "\\x01\" for Plutus V1 scripts
+;   "\\x02\" for Plutus V2 scripts
+;   "\\x03\" for Plutus V3 scripts   
+; 
```

Comment:
This seems to be relevant to Conway, not Alonzo

Reply:
If one intends to override just the comment, one has to redefine the rule in the current module. So I put the comment in `shelley` for now. I think it is okay, since we already know what `plutus v3` is now. But, I will open an issue in `cuddle` to make comments override-able as well as append-able to.